File: android/src/main/kotlin/com/fit321/fitui/components/FitProfileHero.kt
package com.fit321.fitui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PhotoCamera
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fit321.fitui.tokens.FitColors
import com.fit321.fitui.tokens.FitFont

// ============================================================================
// FitProfileHero — 16:9 hero block with 3-variant media fallback (video /
// cover image / brand-gradient + initials) and optional camera-overlay
// edit affordance. Mirrors Swift `FitProfileHero`.
// See `docs/components.md` § FitProfileHero.
// ============================================================================

sealed class FitProfileHeroMedia {
    data class Video(val url: String) : FitProfileHeroMedia()
    data class Cover(val url: String) : FitProfileHeroMedia()
    data class Initials(val text: String) : FitProfileHeroMedia()
}

@Composable
fun FitProfileHero(
    media: FitProfileHeroMedia,
    onEdit: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .aspectRatio(16f / 9f)
            .background(Color.Black)
    ) {
        when (media) {
            is FitProfileHeroMedia.Video -> VideoPlaceholder()
            is FitProfileHeroMedia.Cover -> CoverPlaceholder()
            is FitProfileHeroMedia.Initials -> InitialsFallback(media.text)
        }

        if (onEdit != null) {
            Box(
                modifier = Modifier
                    .align(Alignment.TopEnd)
                    .padding(top = 12.dp, end = 12.dp)
                    .size(36.dp)
                    .clip(CircleShape)
                    .background(Color.Black.copy(alpha = 0.5f))
                    .clickable { onEdit() },
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Default.PhotoCamera,
                    contentDescription = "Edit cover",
                    tint = Color.White,
                    modifier = Modifier.size(18.dp)
                )
            }
        }
    }
}

@Composable
private fun VideoPlaceholder() {
    // Real impl: ExoPlayer. Prototype-level placeholder = dark gradient + play glyph.
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(16f / 9f)
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        Color(0xFF1A1A2E),
                        Color(0xFF16213E)
                    )
                )
            ),
        contentAlignment = Alignment.Center
    ) {
        Icon(
            Icons.Default.PlayArrow,
            contentDescription = null,
            tint = Color.White.copy(alpha = 0.8f),
            modifier = Modifier.size(48.dp)
        )
    }
}

@Composable
private fun CoverPlaceholder() {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(16f / 9f)
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        Color(0xFF2D5A3D),
                        Color(0xFF4A7C59)
                    )
                )
            )
    )
}

@Composable
private fun InitialsFallback(initials: String) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(16f / 9f)
            .background(FitColors.brandGradient),
        contentAlignment = Alignment.Center
    ) {
        Text(
            initials.take(2).uppercase(),
            style = FitFont.heading1.copy(
                fontSize = 56.sp,
                fontWeight = FontWeight.SemiBold,
                letterSpacing = 2.sp
            ),
            color = Color.White.copy(alpha = 0.95f)
        )
    }
}

File: Sources/FitUI/Components/FitProfileHero.swift
import SwiftUI

// MARK: - FitProfileHero
//
// 16:9 hero media block at the top of the coach profile. Renders in
// 3 fallback variants:
//   .video(url)    — preferred; native player
//   .cover(url)    — secondary; image fill
//   .initials(s)   — fallback; brand-gradient + 56pt initials
// Optional onEdit callback renders a camera-overlay button (top-right)
// that the caller wires to a media picker.
// See `docs/components.md` § FitProfileHero.

public struct FitProfileHero: View {
    public enum Media {
        case video(URL)
        case cover(URL)
        case initials(String)
    }

    let media: Media
    let onEdit: (() -> Void)?

    @Environment(\.fitTheme) private var theme

    public init(media: Media, onEdit: (() -> Void)? = nil) {
        self.media = media
        self.onEdit = onEdit
    }

    public var body: some View {
        ZStack(alignment: .topTrailing) {
            content
                .frame(maxWidth: .infinity)
                .aspectRatio(16.0 / 9.0, contentMode: .fit)
                .background(Color.black)
                .clipped()

            if let onEdit {
                Button(action: onEdit) {
                    Image(systemName: "camera.fill")
                        .font(.system(size: 16))
                        .foregroundColor(.white)
                        .frame(width: 36, height: 36)
                        .background(
                            Color.black.opacity(0.5),
                            in: Circle()
                        )
                }
                .buttonStyle(.plain)
                .padding(.top, 12)
                .padding(.trailing, 12)
            }
        }
    }

    @ViewBuilder
    private var content: some View {
        switch media {
        case .video(let url):
            // Real impl: native AVPlayer / AVPlayerLayer. Prototype-level
            // placeholder showing play glyph on dark gradient.
            ZStack {
                LinearGradient(
                    colors: [Color(red: 0.1, green: 0.1, blue: 0.18),
                             Color(red: 0.09, green: 0.13, blue: 0.24)],
                    startPoint: .topLeading, endPoint: .bottomTrailing
                )
                Image(systemName: "play.fill")
                    .font(.system(size: 48))
                    .foregroundColor(.white.opacity(0.8))
            }
        case .cover(let url):
            AsyncImage(url: url) { phase in
                switch phase {
                case .success(let image):
                    image.resizable().scaledToFill()
                default:
                    LinearGradient(
                        colors: [Color(red: 0.18, green: 0.35, blue: 0.24),
                                 Color(red: 0.29, green: 0.49, blue: 0.35)],
                        startPoint: .topLeading, endPoint: .bottomTrailing
                    )
                }
            }
        case .initials(let initials):
            ZStack {
                FitColors.brandGradient
                Text(initials.prefix(2).uppercased())
                    .font(.custom(FitFont.family, size: 56).weight(.semibold))
                    .foregroundColor(.white.opacity(0.95))
                    .tracking(2)
            }
        }
    }
}
